### DISCUSSION TITLE
How do I go about testing with Prisma?

### DISCUSSION THREAD

**Discussion by olliepotter** [prisma/prisma#5262] [Q&A] [answered]
Hi, I'm trying to write my test suite with mocha and chai in node. I'm getting to grips with it and things are going well, now I'm trying to auto-wipe and seed my database with test data at the start of my suite...

Is there a standard way of doing this or any tools that can help?

**Comment by ryands17** [comment, marked as answer]
Hey @olliepotter 👋 

Currently we have an example of testing [here](https://github.com/prisma/prisma-examples/tree/latest/typescript/testing-express) with SQLite and [one](https://github.com/ryands17/nexus-introspection) with Postgres where you can wipe the database and seed it after running the migrations as mentioned [here](https://www.prisma.io/docs/guides/prisma-guides/seed-database).

Let me know if that helps :)

**Reply by olliepotter** [reply]
Hey @ryands17 👋,

Firstly, thanks for replying! Really appreciate it. So I've downloaded the repo and just running through the readme instructions... I've renamed the .env file but when trying to hit an endpoint it says the env variable 'DB_URL' is not set so it obviously hasn't been loaded. I looked into the scripts and I actually can't see anywhere that the env file is loaded at all? Am I missing something here? I'm going to install dotenv now and modify the script to load it but just wanted to let you know or see if there is something I'm missing. :)

**Reply by ryands17** [reply]
Prisma automatically looks for the `.env` file in the root of your workspace, so that should work out of the box. What command did you run which caused this error?

**Reply by olliepotter** [reply]
Just did an `npm run dev` then tried to do GET /users. 

For context, I'm running on Windows 10 and I also checked my `process.env` and the variable hadn't been loaded.

**Reply by olliepotter** [reply]
Adding `require('dotenv').config();` and the dotenv package fixed it for me as expected.

**Reply by ryands17** [reply]
It won't be in the process.env as Prisma handles that automatically. Could you try running `prisma generate` and try again?

**Reply by olliepotter** [reply]
Weirdly yeah it seems to be all good now? (I completely removed dotenv to be sure)

**Reply by olliepotter** [reply]
So I can see how the testing is done here for an SQLite DB as it's obviously easy to create a file and test there then delete the db afterwards... I'm using MySQL as my driver and I'm still a bit unsure as to how I would go about this? Do I need to get prisma to spin up a separate schema on my DB then drop it afterwards? Is this possible?

**Reply by ryands17** [reply]
Yes, I have done the same in the example I've shared above that uses Postgres.

**Reply by olliepotter** [reply]
Ahh yes, I should have looked at that one to begin with. Awesome! Thanks, I have it working just needed to modify the test environment and all was sorted. Loving Prisma, thanks for your help @ryands17 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
